lib/winapi/sync.c

```diff
@@ -40,6 +40,106 @@ VOID LeaveCriticalSection (LPCRITICAL_SECTION lpCriticalSection)
     RtlLeaveCriticalSection(lpCriticalSection);
 }
 
+void AcquireSRWLockExclusive (PSRWLOCK SRWLock)
+{
+    // Grab the global lock (second-highest bit)
+    uintptr_t unlocked_val = SRWLock->Ptr & 0xBFFFFFFF;
```

Comment:
This does the opposite to what the comment says? It "grabs" (poor word choice) all bits, except the second highest bit.

nit: I'd prefer `~0x40000000` or `~(1 << 30)`

Reply:
The comment refers to all of the following lines, which grab the lock by trying to set the bit.